packages/foam-core/src/note-graph.ts

```diff
@@ -15,17 +15,28 @@ export interface NoteLink {
   position: Position;
 }
 
+export interface NoteLinkDefinition {
+  label: string;
+  url: string;
+  title?: string;
+  position?: Position;
+}
+
 export class Note {
   public id: ID;
   public title: string;
   public source: string;
   public path: string;
+  public end: Point;
```

Comment:
FYI @riccardoferretti @jojanaho I added Note.end (the line/col/offset at end of file) and Note.definitions (the existing parsed link ref defs and their positions) here. 

Ideally, there would be a cleaner way of doing this. e.g. some kind of metadata container on the graph node (and arguably, the definitions should live on the edges, maybe?). I'm open to refactoring the naming/storage of these, but any conceptual issues with this?

Reply:
It's the definitions that already exist in the file. `end` is the line/col/offset at end of file -- essentially where the markdown AST root node ends, and new content may be appended

Reply:
@jevakallio if I understand correctly, you are now taking into account the existing definitions within the file (I guess so that we don't double-generate them?).
Do you think we could just treat our generated code as regular definitions? What I mean is that instead of having a defined (code-lensed) section for the definition, we simply generate definitions at the bottom of the file for any missing wikilink.

- the advantage is that it integrates the 2 use cases
- might even automatically solve #94 
- possible drawback is that it might be a bit spooky to generate text in the file without properly flagging it
- (meta: this might not be the best place to discuss, happy to take it to a different PR/issue)

Reply:
Note: above conversation led to proposal https://github.com/foambubble/foam/pull/122